test/pkg/environment/aws/expectations.go

```diff
@@ -115,24 +115,20 @@ func (env *Environment) ExpectExperimentTemplateDeleted(id string) {
 	Expect(err).ToNot(HaveOccurred())
 }
 
-func (env *Environment) EventuallyExpectInstanceProfileExists(profileName string) {
+func (env *Environment) EventuallyExpectInstanceProfileExists(profileName string) iam.InstanceProfile {
```

Comment:
Does this still need to be an Eventually check or can we be assured that the creation of the instance profile is effectively synchronous and we shouldn't see issues with calls to RunInstances

Reply:
The reason I cut the initial PR was that there's a chance for there to be a delay. I don't think an eventually hurts here, just makes it more likely that we don't fail from a fleeting error.